address_book/internal/service/service.go

```diff
@@ -0,0 +1,126 @@
+package service
+
+import (
+	"addressbook/internal/pb"
+	"context"
+	"fmt"
+	"regexp"
+	"sync"
+)
+
+type AddressBookService struct {
+	pb.AddressBookServiceServer
+	data sync.Map
+}
+
+func (a *AddressBookService) AddContact(_ context.Context, in *pb.AddContactRequest) (*pb.AddContactResponse, error) {
+	if in == nil || in.Contact == nil {
+		return &pb.AddContactResponse{
+			Msg: "Contact has not been added",
+		}, nil
+	}
+
+	a.data.Store(in.Contact.Phone, in.Contact)
+
+	return &pb.AddContactResponse{
+		Msg: "Contact added successfully",
+	}, nil
+}
+
+func (a *AddressBookService) FindContactByName(_ context.Context, in *pb.FindContactByNameRequest) (*pb.FindContactResponse, error) {
+	if in == nil {
+		return &pb.FindContactResponse{
+			Msg: "Contact has not been found",
+		}, nil
+	}
+
+	contacts := []*pb.Contact{}
+
+	a.data.Range(func(key interface{}, value interface{}) bool {
+		if contact, ok := value.(*pb.Contact); ok {
+			if contact.Name == in.Name {
+				contacts = append(contacts, contact)
+			}
+		}
+		return true
+	})
+
+	return &pb.FindContactResponse{
+		Contacts: contacts,
+		Msg:      getFindContactMsg(len(contacts)),
+	}, nil
+}
+
+func (a *AddressBookService) FindContactByPhone(_ context.Context, in *pb.FindContactByPhoneRequest) (*pb.FindContactResponse, error) {
```

Comment:
I suggest using 1 endpoint like FindContact and use FindContactByPhone FindContactByName inside the FindContact
it will be more convenient for user

Reply:
FindContact endpoint has been updated